### ISSUE TITLE
WCAG 1.4.10 Reflow – Content partially cut off at 400% zoom

### ISSUE THREAD

**Issue by alunaticity** [hypothesis/product-backlog#1660]
Sidebar content was cut off when the browser zoom was set to 400%. 

![Image](https://github.com/user-attachments/assets/82764215-57b4-4b74-a187-33b8c714e3d7)

This was previously marked as "Supports" in the 2024 VPAT, but the 2025 review included a screenshot demonstrating the issue.

**Comment by robertknight** [comment]
The WCAG 1.4.10 requirement is not 400% zoom specifically. What it actually says is:

<img width="885" height="588" alt="Image" src="https://github.com/user-attachments/assets/32881e05-90ca-4f3a-8c27-1c5ec9b406b8" />

The 320 pixel requirement presumably comes from mobile phones at the time when WCAG AA was originally drafted.

**Comment by robertknight** [comment]
Despite the technicality in the previous comment, the is obviously an issue that is worth fixing. We need a design for what the top bar should look like given the combination of long group name, narrow width and touch mode. Do we use two rows? Replace the group name with an icon? Something else?

**Comment by acelaya** [comment]
I think there's actually a couple of different issues here, which may require different solutions.

1. The first one is the general lack of horizontal space due to the sidebar constraints. That can cause the group name to be too long to fit the available space (regardless of the zoom level), specially in the LMS, since in the web app we limit group names to 25 characters.
2. Zooming-in can affect the elements arrangement, which again can cause the group name to be too long to fit. However, due to previous point, the space issue is present at any zoom level.
3. Touch devices  will usually have a lower screen resolution, which adds up to previous two problems.

As Rob mentions above, we need a design that accounts for these issues and defines how the sidebar toolbar and group dropdown should behave when the space is not enough to contain everything.

One possibility would be to just display a group icon, or truncate the group name to the available space with an ellispis:

<img width="406" height="78" alt="Image" src="https://github.com/user-attachments/assets/8cfc0c6d-ff3f-4b71-bc39-47eb021e398d" />

The group has a title which is displayed on hover, so any of those two would solve points 1 and 2, but you cannot hover on a touch device, so we would still need something for that case.

<img width="487" height="77" alt="Image" src="https://github.com/user-attachments/assets/93b6f4f5-2311-4281-9493-d67154020ece" />

If the originally reported issue was related with the zoom, that might be a solution good enough, and we could address the issue with touch devices separately.

**Comment by mkdir-washington-edu** [comment]
**Truncation**
I think truncating the name with ellipses, and having the full name available on hover, makes sense when we lack horizontal space (whether due to zoom or other reasons). Just a note that the LMS app group names as we store them in the "group" table in H are also truncated to 25 chars, I believe.

On mobile, could the full name be revealed on a long press of whatever's left of the title?

**Group icons**
I think eliminating the text and only using group icons could be confusing in situations where the user can see multiple groups. Plus we currently only show icons for the first group in an organization (like we do in the web app), so I think we'd have to change the behavior of the icons as it is now:

<img width="261" height="167" alt="Image" src="https://github.com/user-attachments/assets/9732bac5-6009-4850-a65b-b0e90ee6eef9" />

Therefore I think when truncating we should actually consider first dropping the icon, then truncating the text.

**Other options**
While I want to solve this without making the scope huge, I wonder if we could get some group name space back by putting the search, order, share, support, account icons above the group name when there's not enough space in the Sidebar, Though it's possible that as we lose horizontal space we'll also lose vertical space, making it undesirable to gain yet another row information that isn't annotation Cards.


To make forward, when we run out of horizontal space, I would:
- Lose the group icon
- Truncate group name text as needed
- Have the group name text visible on hover
- If possible, on mobile have the group name text visible on a long press or similar.

If you need a more formal design let me know.

**Comment by acelaya** [comment]
> If you need a more formal design let me know.

Thanks! I think this is more than enough. 

I'll do a bit of experimentation and see what I can get. I'll probably get individual fixes merged independently and link all PRs to this issue.

**Comment by acelaya** [comment]
This is a summary of all the changes we have applied to address this:

* The group name no longer wraps multiple lines when it's too long. Instead, the text is truncated with an ellipsis.
* The full group name is still available on hover, via `title` attribute, which screen readers announce (this is actually not new, but works well with the point above).
* Hide the group logo when the sidebar is smaller than 350px, to leave the group name as much space as possible.

We have discussed here a few extra points to improve the experience in touch devices:

* Show the full group name on long-press (or similar), since there's no "hover" in that context (https://github.com/hypothesis/client/issues/7314).
* Wrap group selector and sidebar toolbar into two rows in low resolutions, to maximize space for the group name (https://github.com/hypothesis/client/issues/7315).

I reckon these two things might be worth addressing eventually, but I think they are technically out of the scope of the originally reported issue.

Because of that, I'm going to close this issue as done, and we can decide to work on those touch device improvements separately.
